TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Domain/Model/AbstractNodeData.php

```diff
@@ -213,7 +213,7 @@ public function hasProperty($propertyName)
         if (is_object($this->contentObjectProxy)) {
             return ObjectAccess::isPropertyGettable($this->contentObjectProxy->getObject(), $propertyName);
         }
-        return isset($this->properties[$propertyName]);
+        return isset($this->properties[$propertyName]) || array_key_exists($propertyName, $this->properties);
```

Comment:
As I already admitted in https://github.com/neos/flow-development-collection/pull/686:

> You're probably right. This is way too much micro-optimization. With PHP 5.6 array_key_exists() is about three times slower than isset() but with PHP 7 it's way closer ;)